### ISSUE TITLE
target layer for regression

### ISSUE THREAD

**Issue by JaeseungYeom** [LLNL/lbann#198]
In the line 92 of include/lbann/layers/io/target/generic_target_layer.hpp
```
  void setup_dims() override {
    io_layer::setup_dims();
    if (this->is_for_regression()) {
      this->m_neuron_dims = get_data_dims();
      this->m_num_neuron_dims = this->m_neuron_dims.size();
      this->m_num_neurons = std::accumulate(this->m_neuron_dims.begin(),
                                            this->m_neuron_dims.end(),
                                            1,
                                            std::multiplies<int>());
    } else {
      this->m_num_neurons = get_linearized_label_size();
      this->m_num_neuron_dims = 1;
      this->m_neuron_dims.assign(1, this->m_num_neurons);
    }
  }```
Why are we using get_data_dims()? Shouldn't it be
```
    if (this->is_for_regression()) {
      this->m_num_neurons = get_linearized_response_size();
      this->m_num_neuron_dims = 1;
      this->m_neuron_dims.assign(1, this->m_num_neurons);
    } else {
```
Otherwise, do we need generic_data_reader::get_response_dims() ?

**Comment by JaeseungYeom** [comment]
Based on the discussion in the last meeting, it looks like that regression has not been used recently. I think this is a bug.
Together with this, generic_input_layer::get_linearized_response_size() needs to be fixed, We also needs to set m_for_regression to the io_layer as the base of an input layer. Therefore, we need the prototext support to set it for the input layer as we have for the target layer. Then, the prototext parser needs to check if the paired input layer of a target layer is consistent on this. I will create a PR.

**Comment by timmoon10** [comment]
I think we may be able to delete the entire `setup_dims` function. Target layers have no children, so their `num_neurons` variable is not meaningful.

**Comment by JaeseungYeom** [comment]
fixed in commit  85247aa (PR 202)
